File: task3/go/SubTask4.go
package main

import (
    "fmt"
    "os"
)

func printPatternToFile(n int, outputFile *os.File) {
    for i := 1; i <= (n+1)/2; i++ {
        for j := 0; j < (n+1)/2-i; j++ {
            fmt.Fprint(outputFile, " ")
            fmt.Print(" ")
        }
        for j := 0; j < (i*2)-1; j++ {
            fmt.Fprint(outputFile, "*")
            fmt.Print("*")
        }
        fmt.Fprintln(outputFile)
        fmt.Println()
    }

    for i := (n+1)/2 + 1; i <= n; i++ {
        for j := 0; j < i-(n+1)/2; j++ {
            fmt.Fprint(outputFile, " ")
            fmt.Print(" ")
        }
        for j := 0; j < (n+1-i)*2-1; j++ {
            fmt.Fprint(outputFile, "*")
            fmt.Print("*")
        }
        fmt.Fprintln(outputFile)
        fmt.Println()
    }
}

func main() {
    var n int
    fmt.Print("Enter the number of rows: ")
    fmt.Scan(&n)

    outputFile, err := os.Create("output.txt")
    if err != nil {
        fmt.Println("Error opening output.txt")
        return
    }
    defer outputFile.Close()

    printPatternToFile(n, outputFile)
    fmt.Println("Diamond pattern has been written to output.txt")
}

File: task3/rust/Sub-Task4.rs
use std::fs::File;
use std::io::{self, Write};

fn print_pattern_to_file(n: i32, output_file: &mut File) {
    for i in 1..=(n + 1) / 2 {
        for _ in 0..(n + 1) / 2 - i {
            write!(output_file, " ").unwrap();
            print!(" ");
        }
        for _ in 0..(i * 2) - 1 {
            write!(output_file, "*").unwrap();
            print!("*");
        }
        writeln!(output_file).unwrap();
        println!();
        io::stdout().flush().unwrap();
    }

    for i in (n + 1) / 2 + 1..=n {
        for _ in 0..i - (n + 1) / 2 {
            write!(output_file, " ").unwrap();
            print!(" ");
        }
        for _ in 0..(n + 1 - i) * 2 - 1 {
            write!(output_file, "*").unwrap();
            print!("*");
        }
        writeln!(output_file).unwrap();
        println!();
        io::stdout().flush().unwrap();
    }
}

fn main() {
    let mut n = String::new();
    println!("Enter the number of rows: ");
    io::stdin().read_line(&mut n).unwrap();
    let n: i32 = n.trim().parse().unwrap();

    let mut output_file = File::create("output.txt").expect("Error opening output.txt");

    print_pattern_to_file(n, &mut output_file);

    println!("Diamond pattern has been written to output.txt");
}
